### PR TITLE
fix: exclude supplier sourced items while creating SCR from SCO

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
exclude supplier sourced items while creating SCR from SCO
Fetch Supplied items from Subcontracting Order instead of BOM while creating Subcontracting Receipt

Clickup - https://app.clickup.com/t/862jxdy8h

### CODE REVIEW

File: erpnext/controllers/subcontracting_controller.py [hunk 1]
```diff
@@ -395,6 +395,24 @@ def __get_materials_from_bom(self, item_code, bom_no, exploded_item=0):
...
+			"reserve_warehouse",
+			"stock_uom",
+			"sourced_by_supplier",
+			"required_qty",
+			"sourced_by_hyper",
+			"required_qty as qty_consumed_per_unit"
+		]
+
+		return (
+			frappe.get_all("Subcontracting Order Supplied Item", fields=fields, filters={"parent": sco, "main_item_code":item_code, "sourced_by_supplier": 0}) or []
```
Comment (Alchez) [inline erpnext/controllers/subcontracting_controller.py]: `frappe.get_all` will return an empty array by default, do we need the `or []` at the end?

File: erpnext/controllers/subcontracting_controller.py [hunk 2]
```diff
@@ -395,6 +395,24 @@ def __get_materials_from_bom(self, item_code, bom_no, exploded_item=0):
 			frappe.get_all("BOM", fields=fields, filters=filters, order_by=f"`tab{doctype}`.`idx`") or []
 		)
 
+	def __get_supplied_items_from_sco(self, sco, item_code):
+		fields =  [
```
Comment (Alchez) [inline erpnext/controllers/subcontracting_controller.py]:
```suggestion
		fields = [
```

File: erpnext/controllers/subcontracting_controller.py (hunk 1 above)
Comment (Alchez) [inline erpnext/controllers/subcontracting_controller.py]:
```suggestion
			frappe.get_all("Subcontracting Order Supplied Item", fields=fields, filters={"parent": sco, "main_item_code": item_code, "sourced_by_supplier": 0})
```

File: erpnext/controllers/subcontracting_controller.py [hunk 3]
```diff
@@ -515,14 +533,22 @@ def __set_supplied_items(self):
...
-					row.item_code, row.bom, row.get("include_exploded_items")
-				):
-					qty = flt(bom_item.qty_consumed_per_unit) * flt(row.qty) * row.conversion_factor
-					bom_item.main_item_code = row.item_code
-					self.__update_reserve_warehouse(bom_item, row)
-					self.__set_alternative_item(bom_item)
-					self.__add_supplied_item(row, bom_item, qty)
+				if self.doctype == "Subcontracting Receipt" and row.subcontracting_order:
+					for sco_supplied_item in self.__get_supplied_items_from_sco(row.subcontracting_order, row.item_code):
+						qty = flt(sco_supplied_item.qty_consumed_per_unit) * row.conversion_factor
```
Comment (Alchez) [inline erpnext/controllers/subcontracting_controller.py]: Does this need conversion? Or should it be the same as the SCO table?

**Review by Alchez — CHANGES_REQUESTED** [review]

File: erpnext/controllers/subcontracting_controller.py (hunk 3 above)
Comment (MyuddinKhatri) [reply erpnext/controllers/subcontracting_controller.py]: It can be same as SCO table

**Review by Alchez — APPROVED** [review]

**Pull request merged**